packages/mask/src/plugins/Tips/hooks/useKv.ts

```diff
@@ -13,11 +13,12 @@ const getCurrentPersonaPublicKey = async () => {
     if (!currentPersona || !currentPersona.publicHexKey) return ''
     return currentPersona.publicHexKey
 }
-export function useKvGet() {
+export function useKvGet<T>(publicKey: string | null | undefined) {
```

Comment:
This generics does not make sense and its type is unsafe.

Reply:
We can assume `useKvGet<ThisWouldBeWhatWeWant>()`

Reply:
You should not assume. You should check if the data structure is the expected shape in JavaScript and handle the falling case.

Reply:
Well, we store plugin-specified data to kv-service and fetch with this hook, we know what we want for specified plugin

```diff
-export interface NextIDStorageInfo {
+export interface NextIDStorageInfo<T = unknown> {
     persona: string
-    proofs: NextIDStorageProofs<unknown>[]
+    proofs: NextIDStorageProofs<T>[]
 }
 export interface NextIDStorageProofs<T> {
     content: {
         // pluginId
         [index: string]: T
     }
     identity: string
     platform: NextIDPlatform
 }

```

Reply:
So, I think `useKvGet` the name makes people misunderstand. I will understand it is related to `kv` storage. But in fact it is just to get `publicKey` from `kv`